api/v0/proto/retrievability/types.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.types;
+
+// TODO: oneof different blob versions
+message Blob {
+  // The blob data.
+  bytes data = 1;
+  // The signer of the blob.
+  optional bytes signer = 2;
+  // The blob commitment.
+  bytes commitment = 3;
+}
+
+// Blob with blob proof. Either the blob and the proof are optional, depending
+// on the options of the request.
+message BlobWithProof {
+  // The blob.
+  optional Blob blob = 1;
+  // The proof of blob inclusion or non-inclusion.
+  optional DataRootProof proof = 2;
+  // Index of the first share of the blob in the ODS.
+  uint64 index = 3;
+}
+
+// A single data root proof can span multiple rows.
+message DataRootProof {
+  // The type of the proof.
+  ProofType type = 1;
+  // The proof.
+  Proof proof = 2;
+}
+
+enum ProofType {
+  PROOF_TYPE_UNSPECIFIED = 0;
+  // Blob inclusion proof.
+  PROOF_TYPE_INCLUSION = 1;
+  // Blob non-inclusion proof.
+  PROOF_TYPE_NON_INCLUSION = 2;
+}
+
+message Proof {
+  // The range in the ODS. Can span multiple rows.
+  Range range = 1;
```

Comment:
Can't use ODS range for merklee inclusion proof of data. It **must** be EDS range

Reply:
Correction: I've realised it can be, just would be messy painful in verification libs, but thats our problem. For the love of users I'm in favour of keeping it ODS now. (can be resolved)

Reply:
Per @Ferret-san, we can add a method to get the subtree root of the parity shares subtree. That should cover the cases that need access to that inner node for proofs.

Reply:
Added in c1e0f9d427410c1057c743e7ba8e085e97f1e97d